pkgs/by-name/ni/nixos-rebuild-ng/src/tests/test_models.py

```diff
@@ -4,12 +4,21 @@
 from typing import Any
 from unittest.mock import patch
 
+import nixos_rebuild as nr
 import nixos_rebuild.models as m
 
 from .helpers import get_qualified_name
 
-
-def test_build_attr_from_arg() -> None:
+@patch(get_qualified_name(nr.process.subprocess.run), autospec=True)
+def test_build_attr_from_arg(mock_run: Any, tmp_path: Path) -> None:
+    nixpkgs_path = tmp_path / "nixpkgs"
+    mock_run.side_effect = [
+        # with config.nixpkgs.source
+        subprocess.CompletedProcess([], 0, f"\"{nixpkgs_path}\"\n"),
+        # without config.nixpkgs.source
+        subprocess.CompletedProcess([], 1, ""),
+    ]
+    assert m.BuildAttr.from_arg(None, None) == m.BuildAttr(str(nixpkgs_path / "nixos"), None)
     assert m.BuildAttr.from_arg(None, None) == m.BuildAttr("<nixpkgs/nixos>", None)
```

Comment:
I think it is better to do a scoped patch here instead:
```suggestion
def test_build_attr_from_arg(tmp_path: Path) -> None:
    nixpkgs_path = tmp_path / "nixpkgs"
    with patch(get_qualified_name(nr.process.subprocess.run), autospec=True, return_value=subprocess.CompletedProcess([], 0, f"\"{nixpkgs_path}\"\n")):
        assert m.BuildAttr.from_arg(None, None) == m.BuildAttr(str(nixpkgs_path / "nixos"), None)
    with patch(get_qualified_name(nr.process.subprocess.run), autospec=True, return_value=subprocess.CompletedProcess([], 1, "")):
    assert m.BuildAttr.from_arg(None, None) == m.BuildAttr("<nixpkgs/nixos>", None)
```

Reply:
Done, thank you!